pkg/skaffold/deploy/deploy_mux.go

```diff
@@ -161,6 +161,12 @@ func (m DeployerMux) Dependencies() ([]string, error) {
 }
 
 func (m DeployerMux) Cleanup(ctx context.Context, w io.Writer, dryRun bool, manifestsByConfig manifest.ManifestListByConfig) error {
+	// Reverse order of deployers for cleanup to ensure resources
+	// are removed before their definitions are removed.
+	revDeployers := m.deployers
+	for i, j := 0, len(revDeployers)-1; i < j; i, j = i+1, j-1 {
+		revDeployers[i], revDeployers[j] = revDeployers[j], revDeployers[i]
+	}
```

Comment:
Thanks. Was wondering if there is a utility method which already does that.

Reply:
That's a good point. Searching the internet I did not find any lib that does this. I was also supprised the slice type has no internal method since this is such a common task. Searching the skaffold source code for `for i, j` and for `for .*,.*:=.*,len` does not give any results.

The golang completion on the other side provides a `slice.reverse!` method which automatically creates the for loop. Maybe that's why there is no lib or function.

I can put such a function somewhere in the skaffold source code if you wish. I need help finding the location where such a function would be placed best.

Reply:
You can write this function here:
https://github.com/GoogleContainerTools/skaffold/blob/main/pkg/skaffold/util/util.go

Reply:
Looking into this now.

Reply:
@gsquared94 I tried adding a function to the file you suggested. However when I did this, I created a dependency lopp between util.go and deploy.go.

I also tried create a generic slice reverse function in utils but since the go.mod file says to use at least go 1.17, I could not use generics.

In the end I added it to DeployMux since this is the only place using this function for now.